### PR TITLE
Add setRequestResponseTimeout convenience function

### PR DESCRIPTION
**Pull request by MaxGabriel**
I like to use the convenience functions, and it's been annoying this one is missing.

I haven't tested this in practice, but this seems pretty straightforward too?

### CODE REVIEW

File: http-conduit/ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # ChangeLog for http-conduit
 
+## 2.3.7.5
+
+* Adds a convenience function to set a request's response timeout []()
```
Comment (MaxGabriel) [inline http-conduit/ChangeLog.md:5]:
```suggestion
* Adds a convenience function to set a request's response timeout [#456](https://github.com/snoyberg/http-client/pull/456)
```

File: http-conduit/http-conduit.cabal [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cabal-version:   >= 1.10
 name:            http-conduit
-version:         2.3.7.4
+version:         2.3.7.5
```
Comment (MaxGabriel) [inline http-conduit/http-conduit.cabal:3]: Oh, now that I think of it, this should probably be 2.3.8 since this is adding a new API

Comment (snoyberg) [reply http-conduit/http-conduit.cabal:3]: If you can make that change, I'd be happy to merge. Thanks!

Comment (MaxGabriel) [reply http-conduit/http-conduit.cabal]: Cool, updated!

**Review by snoyberg — APPROVED** [review]

**Pull request merged**